internal/goproxy/dashc.go

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package goproxy
+
+import (
+	"os"
+	"strings"
+
+	"github.com/datadog/orchestrion/internal/log"
+)
+
+// ProcessDashC gets the command line arguments passed to a "go" command (without "go" itself), and processes the "-C"
+// flag present at the beginning of the slice (if present), changing directories as requested, then returns the slice
+// without it.
+//
+// The "-C" flags is reuqired to be the very first argument provided to "go" commands.
+func ProcessDashC(args []string) ([]string, error) {
```

Comment:
You can probably use this to fix #161

Reply:
Yah intentionally left for a future PR because I think it might be a little trickier than you think ^^